source/Grid/defaultCellRangeRenderer.js

```diff
@@ -40,7 +97,8 @@ export default function defaultCellRangeRenderer ({
         isScrolling,
         key,
         rowIndex,
-        style
+        style,
+        visibility: findOverlap(gridDimensions, style)
```

Comment:
IMO this calculation is unnecessarily expensive. I don't think it's important to know _how_ visible a cell is, only whether it's visible (at all) or not. Scroll events fire quickly and often. This is probably the single area where perf matters the most in react-virtualized.